lib/requester/core.js

```diff
@@ -289,6 +291,20 @@ var dns = require('dns'),
         // - localhost
         // - *.localhost
         if (getTLD(lowercaseHost) !== LOCALHOST) {
+            // When options.all is set, the callback expects array of addresses instead of a single address
+            if (options.all) {
+                return _lookup(options, hostLookup, lowercaseHost, function (err, addresses) {
+                    if (err) { return callback(err); }
+
+                    addresses = _.filter(addresses, (addr) => {
+                        return !self.isAddressRestricted(addr, networkOpts);
+                    });
+
+                    return callback(_.isEmpty(addresses) ?
```

Comment:
For error why are we checking if addresses exist or not?

Reply:
For a host name if no ip is available (empty list of addresses) we send `Error(ERROR_ADDRESS_RESOLVE + hostname)` ( `NETERR: getaddrinfo ENOTFOUND`)